fma() in GL core profile

I have some shaders written in GLSL version 450 that are being reflected back to GL core profile version 3.2. It seems fma() wasn't added until version 4.0 but is being used in the reflected shader output:

```
[INFO] [slang]: Compiling shader "C:\Users\...\slang-shaders\anti-aliasing\shaders/smaa/smaa-edge-detection.slang".
[INFO] [slang]: Using render target format R8G8B8A8_UNORM for pass output #0.
[INFO] [slang]: Compiling shader "C:\Users\...\slang-shaders\anti-aliasing\shaders/smaa/smaa-blend-weight-calculation.slang".
[INFO] [slang]: Using render target format R8G8B8A8_UNORM for pass output #1.
[INFO] [slang]: Compiling shader "C:\Users\...\slang-shaders\anti-aliasing\shaders/smaa/smaa-neighborhood-blending.slang".
[INFO] [slang]: Building pass #0 (N/A)
[INFO] [slang]: Reflection
[INFO] [slang]:   Textures:
[INFO] [slang]:      Source (#0)
[INFO] [slang]:
[INFO] [slang]:   Uniforms (Vertex: yes, Fragment: yes):
[INFO] [slang]:   Push Constants (Vertex: yes, Fragment: yes):
[INFO] [slang]:      MVP (Offset: 0)
[INFO] [slang]:      SourceSize (#0) (PushOffset: 0)
[INFO] [slang]:
[INFO] [slang]:   Parameters:
[ERROR] [GLCore]: Failed to compile shader: Vertex shader failed to compile with the following errors:
ERROR: 0:27: error(#202) No matching overloaded function found: fma
ERROR: 0:27: error(#160) Cannot convert from: "const float" to: "default out highp 4-component vector of vec4"
ERROR: 0:28: error(#202) No matching overloaded function found: fma
ERROR: 0:28: error(#160) Cannot convert from: "const float" to: "default out highp 4-component vector of vec4"
ERROR: 0:29: error(#202) No matching overloaded function found: fma
ERROR: 0:29: error(#160) Cannot convert from: "const float" to: "default out highp 4-component vector of vec4"
ERROR: error(#273) 6 compilation errors.  No code generated
```

The shaders can be found here:
https://github.com/libretro/slang-shaders/tree/master/anti-aliasing/shaders/smaa
and the error listed above occurs here:
https://github.com/libretro/slang-shaders/blob/master/anti-aliasing/shaders/smaa/smaa-edge-detection.slang#L28

--- a/reference/opt/shaders/legacy/fragment/fma.legacy.frag
+++ b/reference/opt/shaders/legacy/fragment/fma.legacy.frag
@@ -1,0 +1,13 @@
+#version 100
+precision mediump float;
+precision highp int;
+
+varying highp vec4 vA;
+varying highp vec4 vB;
+varying highp vec4 vC;
+
+void main()
+{
+    gl_FragData[0] = vA * vB + vC;
+}
+

--- a/reference/shaders/legacy/fragment/fma.legacy.frag
+++ b/reference/shaders/legacy/fragment/fma.legacy.frag
@@ -1,0 +1,13 @@
+#version 100
+precision mediump float;
+precision highp int;
+
+varying highp vec4 vA;
+varying highp vec4 vB;
+varying highp vec4 vC;
+
+void main()
+{
+    gl_FragData[0] = vA * vB + vC;
+}
+

--- a/shaders/legacy/fragment/fma.legacy.frag
+++ b/shaders/legacy/fragment/fma.legacy.frag
@@ -1,0 +1,11 @@
+#version 450
+
+layout(location = 0) in vec4 vA;
+layout(location = 1) in vec4 vB;
+layout(location = 2) in vec4 vC;
+layout(location = 0) out vec4 FragColor;
+
+void main()
+{
+   FragColor = fma(vA, vB, vC);
+}

--- a/spirv_glsl.cpp
+++ b/spirv_glsl.cpp
@@ -4818,7 +4818,18 @@ void CompilerGLSL::emit_glsl_op(uint32_t result_type, uint32_t id, uint32_t eop,
 		emit_unary_func_op(result_type, id, args[0], "degrees");
 		break;
 	case GLSLstd450Fma:
-		emit_trinary_func_op(result_type, id, args[0], args[1], args[2], "fma");
+		if ((!options.es && options.version < 400) || (options.es && options.version < 320))
+		{
+			auto expr = join(to_enclosed_expression(args[0]), " * ", to_enclosed_expression(args[1]), " + ",
+			                 to_enclosed_expression(args[2]));
+
+			emit_op(result_type, id, expr,
+			        should_forward(args[0]) && should_forward(args[1]) && should_forward(args[2]));
+			for (uint32_t i = 0; i < 3; i++)
+				inherit_expression_dependencies(id, args[i]);
+		}
+		else
+			emit_trinary_func_op(result_type, id, args[0], args[1], args[2], "fma");
 		break;
 	case GLSLstd450Modf:
 		register_call_out_argument(args[1]);

